src/syntree.rs

```diff
@@ -43,62 +53,102 @@ pub enum Op {
     Div,
 }
 
+/// The different kinds of [Statement]s.
+///
+/// There are both shorter statements like `a = b + 1` as well as longer
+/// statements like `if a { ... } else { ...}`. The variants here include
+/// examples of how they look in the code.
+///
+/// Note that this shouldn't be read as a formal language specification.
 #[derive(Debug, Clone)]
 pub enum StatementKind {
+    /// "Imports" another file.
+    ///
+    /// `use <file>`.
     Use {
         file: Identifier,
     },
 
+    /// Defines a new Blob.
+    ///
+    /// `A :: Blob { <field>.. }`.
     Blob {
         name: String,
         fields: HashMap<String, Type>,
     },
 
+    /// Prints to standard out.
+    ///
+    /// `print <expression>`.
     Print {
         value: Expression,
     },
 
+    /// Assigns to a variable (`a = <expression>`), optionally with an operator
+    /// applied (`a += <expression>`)
     Assignment {
         kind: Op,
         target: Assignable,
         value: Expression,
     },
 
+    /// Defines a new variable.
+    ///
+    /// Example: `a := <expression>`.
+    ///
+    /// Valid definition operators are `::`, `:=` and `: <type> =`.
     Definition {
         ident: Identifier,
         kind: VarKind,
         ty: Type,
         value: Expression,
     },
 
+    /// Makes your code go either here or there.
+    ///
+    /// `if <expression> <statement> [else <statement>]`.
     If {
         condition: Expression,
         pass: Box<Statement>,
         fail: Box<Statement>,
     },
 
+    /// Do something as long as something else evaluates to true.
+    ///
+    /// `loop <expression> <statement>`.
```

Comment:
I'm thinking of joinking `loop { print "infinet loop" }` from rust... Just a heads up.

Reply:
Feel free to edit the doc when that is the case :)